common/Resources/ti.internal/extensions/node/buffer.js

```diff
@@ -0,0 +1,1575 @@
+	 * @returns {Buffer}
+	 */
+	static concat(list, totalLength) {
+		if (!Array.isArray(list)) {
+			throw new TypeError('list argument must be an Array');
+		}
+		if (list.length === 0) {
+			return Buffer.alloc(0); // one empty Buffer!
+		}
+		// allocate one Buffer of `totalLength`? Cap at totalLength?
+		if (totalLength === undefined) {
+			totalLength = 0;
+			// generate the total length from each buffer's length?
+			for (let i = 0; i < list.length; i++) {
+				totalLength += list[i].length;
+			}
+		}
+		const result = Buffer.allocUnsafe(totalLength);
+		let position = 0;
+		for (let i = 0; i < list.length; i++) {
+			const buf = list[i];
+			buf.copy(result, position);
+			position += buf.length;
+			if (position >= totalLength) {
+				break;
+			}
+		}
+		return result;
+	}
+
+	/**
+	 * @param {integer[]|Buffer|string} value value we're wrapping
+	 * @param {string} [encoding='utf8'] The encoding of string.
+	 * @returns {Buffer}
+	 */
+	static from(value, encoding = 'utf8') {
+		const valueType = typeof value;
+		if (valueType === 'string') {
+			if (!Buffer.isEncoding(encoding)) {
+				throw new TypeError(`Unknown encoding: ${encoding}`);
+			}
+			encoding = encoding.toLowerCase();
+			if (encoding === 'base64') {
+				const blob = Ti.Utils.base64decode(value);
+				const blobStream = Ti.Stream.createStream({ source: blob, mode: Ti.Stream.MODE_READ });
+				const buffer = Ti.Stream.readAll(blobStream);
+				blobStream.close();
+				return newBuffer(buffer);
+			}
+			if (encoding === 'hex') {
+				return Buffer.from(stringToHexBytes(value));
+			}
+			return newBuffer(Ti.createBuffer({ value: value, type: getTiCodecCharset(encoding) }));
+		} else if (valueType === 'object') {
+			if (Array.isArray(value)) {
+				const tiBuffer = Ti.createBuffer({ length: value.length });
+				for (let i = 0; i < value.length; i++) {
+					tiBuffer[i] = value[i] & 0xFF; // mask to one byte
+				}
+				return newBuffer(tiBuffer);
+			}
+			if (Buffer.isBuffer(value)) {
+				const length = value.length;
+				const buffer = Buffer.allocUnsafe(length);
+
+				if (length === 0) {
+					return buffer;
+				}
+
+				value.copy(buffer, 0, 0, length);
+				return buffer;
+			}
+			if (value.apiName && value.apiName === 'Ti.Buffer') {
+				return newBuffer(value);
+			}
+		}
+		throw new TypeError('The \'value\' argument must be one of type: \'string\', \'Array\', \'Buffer\', \'Ti.Buffer\'');
+	}
+
+	/**
+	 * @param {string} encoding possible encoding name
+	 * @returns {boolean}
+	 */
+	static isEncoding(encoding) {
+		if (typeof encoding !== 'string') {
+			return false;
+		}
+		return VALID_ENCODINGS.includes(encoding.toLowerCase());
+	}
+
+	/**
+	 * @param {*} obj possible Buffer instance
+	 * @returns {boolean}
+	 */
+	static isBuffer(obj) {
+		return obj !== null && obj !== undefined && obj._isBuffer === true;
+	}
+}
+
+Buffer.poolSize = 8192;
+
+export default Buffer;
```

Comment:
The export is not compatible with the node `buffer` module this way. The core module exports it as `Buffer` instead of default.

Some third-party node modules do the following for example:

```js
var Buffer = require('buffer').Buffer;
```

Reply:
good catch, pushing commits to fix this now